dataland-frontend/src/components/forms/CreateEUTaxonomyForFinancials.vue

```diff
@@ -140,30 +342,163 @@ export default defineComponent({
-  }),
+    reportingPeriod: function (newValue: Date) {
+      this.reportingPeriodYear = newValue.getFullYear();
+    },
+  },
   props: {
     companyID: {
       type: String,
     },
   },
+  mounted() {
+    const dataId = this.route.query.templateDataId;
+
+    if (typeof dataId === "string" && dataId !== "") {
+      this.editMode = true;
+      void this.loadEuData(dataId);
+    }
+
+    this.onThisPageLinks = [...this.onThisPageLinksStart];
+  },
+
   methods: {
+    /**
+     * Loads the Dataset by the provided dataId and pre-configures the form to contain the data
+     * from the dataset
+     *
+     * @param dataId the id of the dataset to load
+     */
+    async loadEuData(dataId: string): Promise<void> {
+      this.waitingForData = true;
+      const euTaxonomyDataForFinancialsControllerApi = await new ApiClientProvider(
+        assertDefined(this.getKeycloakPromise)()
+      ).getEuTaxonomyDataForFinancialsControllerApi();
+
+      const dataResponse =
+        await euTaxonomyDataForFinancialsControllerApi.getCompanyAssociatedEuTaxonomyDataForFinancials(dataId);
+      const dataResponseData = dataResponse.data;
+      if (dataResponseData.data?.fiscalYearEnd) {
+        this.fiscalYearEnd = new Date(dataResponseData.data.fiscalYearEnd);
+      }
+      if (dataResponseData.data?.referencedReports) {
+        const propertiesOfFilesAssignedToDataID = dataResponseData.data.referencedReports;
+        for (const key in propertiesOfFilesAssignedToDataID) {
+          this.uploadFiles.push({
+            name: key,
+            currency: propertiesOfFilesAssignedToDataID[key].currency,
+            isGroupLevel: propertiesOfFilesAssignedToDataID[key].isGroupLevel,
+            reference: propertiesOfFilesAssignedToDataID[key].reference,
+            reportDate: propertiesOfFilesAssignedToDataID[key].reportDate,
+            convertedReportDate: propertiesOfFilesAssignedToDataID[key].reportDate
+              ? new Date(propertiesOfFilesAssignedToDataID[key].reportDate as string)
+              : "",
+          });
+        }
+      }
+      if (dataResponseData.data?.financialServicesTypes) {
+        // types of company financial services
+        const arrayWithCompanyKpiTypes = dataResponseData.data?.financialServicesTypes;
+        // all types of financial services
+        const allTypesOfFinancialServices = this.euTaxonomyKPIsModel.companyTypeToEligibilityKpis;
+
+        this.selectedKPIs = this.kpisModel.filter((el: { label: string; value: string }) => {
+          return arrayWithCompanyKpiTypes?.includes(
+            allTypesOfFinancialServices[
+              el.value as keyof typeof allTypesOfFinancialServices
+            ] as EuTaxonomyDataForFinancialsFinancialServicesTypesEnum
+          );
+        });
+        this.confirmeSelectedKPIs();
+      }
+      const receivedFormInputsModel = modifyObjectKeys(
+        JSON.parse(JSON.stringify(dataResponseData)) as objectType,
+        "receive"
+      );
```

Comment:
I have a feeling that we could make this happen without the `.stringify` and `.parse`.
I mean `dataResponseData` is already a parsed object of type `CompanyAssociatedData...`. And it should be possible to just cast it to `objectType`.
Have you tried just writing:
`const receivedFormInputsModel = modifyObjectKeys(
          dataResponseData as objectType,
        "receive"
      );`

Were you facing issues tryting to do that?

Reply:
Changes were not reflected in the data structure, but I can look deeper into why and if there is a more elegant solution if this conversion looks unacceptable ?

Reply:
Leave it like it is for now.   I can do the research on that to save your time for more important stuff

Reply:
passed to @stewal19

Reply:
parse and stringify removed as they seemingly were not necessary.
observed a bug and created a ticket for it